tests/Service/PromotionsService/FindActivePromotionsByContextTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use DateTimeImmutable;
+
+class FindActivePromotionsByContextTest extends AbstractPromotionServiceTest
+{
+    /** @var CoreEntity */
+    private $context;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $dbId = 1;
+        $context = $this->mockEntity('Brand', $dbId);
+        $context->method('getDbAncestryIds')->willReturn([$dbId]);
+
+        $this->context = $context;
+    }
+
+    public function testFindActivePromotionsByEntityDefaultPagination()
+    {
+        $dbData = [['pid' => 'p000000h']];
+
+        $this->mockRepository
+            ->method('findActivePromotionsByContext')
+            ->with($this->context->getDbAncestryIds(), $this->isInstanceOf(DateTimeImmutable::class), 300, 0)
+            ->willReturn($dbData);
+
+        $promotions = $this->service()->findActivePromotionsByContext($this->context);
+
+        $this->assertEquals($this->getDomainModelsFromDbData($dbData), $promotions);
```

Comment:
Would asserting some other things on the returned value be useful? Such as the number of items returned. Or the value of a specific item rather than just the whole return value equals this whole value?